status-im/status-react
# #15665 Settings list component

Fixes #15665 

### Summary

UI implementation of #15665
Expected design [here](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=1206-28032&t=RIyherjo5sFBK4Rt-0)

Adding settings-list component with initial 6 variants for light and dark theme
Also adding its preview screen

### Review notes
- A prop could be added in the Switch for Android once there is a design, for the moment leave the implementation commented in the code.
- Communities icons are fetched from a url because the png downloaded from Figma were missing a part in the right side.

### Testing notes
Not ready

#### Platforms

- Android
- iOS

#### Areas that maybe impacted

##### Functional

- settings

### Steps to test

- Open Status
- Login
- Tap on profile icon, upper left corner of the screen, from the first or second tabs.
- Tap on `Quo2.0 Preview` option
- Scroll down until `Settings` section is visible and tap on `settings-list button`.
- You should see the preview screen with the component and options to build its variants.

status: wip

### Gifs/MP4


https://github.com/status-im/status-mobile/assets/29315633/36f92d01-2f55-4bb4-9559-666a9570f3e8

https://github.com/status-im/status-mobile/assets/29315633/91ef91a8-9085-4782-a222-758bc3a43556

https://github.com/status-im/status-mobile/assets/29315633/f1176f82-48de-407f-af7e-e356673aef44

https://github.com/status-im/status-mobile/assets/29315633/16ba24ca-d26f-4191-b999-c36543c62d6e

https://github.com/status-im/status-mobile/assets/29315633/6c921613-6a71-4f49-a57e-899d2a2a2a40

https://github.com/status-im/status-mobile/assets/29315633/f0e5c9df-15ed-4b28-9d7e-aa82e9b8e307
